### PR TITLE
Replace urlparse with urlsplit

### PR DESCRIPTION
**Pull request by westonkl**
Replaces [urlparse](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) with [urlsplit](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit) as `urlsplit` is faster and lighterweight.

`urlsplit` is similar to [urlparse()](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse), but does not split the params from the URL. The params functionality allows you to put variables in the url. This should generally be used instead of `urlparse()`.

`urlsplit`: (addressing scheme, network location, path, query, fragment identifier)
`urlparse`: (addressing scheme, network location, path, **params**, query, fragment identifier)

---

### CODE REVIEW

**Comment by BobDu** [comment]
why are you say `urlsplit ` is faster and lighterweight?
python official docs not record it, even, I didn't find any other information recommending use `urlsplit`.

**Comment by WestonKing-Leatham** [comment]
> why are you say `urlsplit` is faster and lighterweight? python official docs not record it, even, I didn't find any other information recommending use `urlsplit`.

Its not much faster and maybe its a bit overselling it to be calling it lighterweight, but it is faster as it doesn't try to parse the url for params. The [second sentence here](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit) of python docs on it says "This should generally be used instead of urlparse()".

**Comment by WestonKing-Leatham** [comment]
Btw there were other files that could have this change applied to them, but I just wanted to see if it was a wanted change before I updated them too.

**Comment by BobDu** [comment]
> > why are you say `urlsplit` is faster and lighterweight? python official docs not record it, even, I didn't find any other information recommending use `urlsplit`.
> 
> Its not much faster and maybe its a bit overselling it to be calling it lighterweight, but it is faster as it doesn't try to parse the url for params. The [second sentence here](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit) of python docs on it says "This should generally be used instead of urlparse()".

Emm,  `if the more recent URL syntax allowing parameters to be applied to each segment of the path portion of the URL (see [RFC 2396](https://datatracker.ietf.org/doc/html/rfc2396.html)) is wanted.`

But, It seems we don't need support RFC 2396 update syntax?

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
I don’t think there’s any perceptible performance difference between `urlsplit` and `urlparse`, but the former is indeed better; literally nobody uses RFC 2396 now.

**Comment by eladkal** [comment]
there are more usages for this in the project. Should we adress them as well?

**Comment by WestonKing-Leatham** [comment]
> there are more usages for this in the project. Should we adress them as well?

Yup, I can add them into this pr

**Comment by BobDu** [comment]
some small ideas, If we replace all `urlparse` to `urlsplit`, can we add some lint rule in pre-commit to prevent new code use `urlparse` again in future.

**Review by dstandish — DISMISSED** [review]
looks like something went wrong with your git operations. you have a lot of unrelated changes.

**Comment by WestonKing-Leatham** [comment]
> 

Yeah I just saw, should I create a temp branch and cherry pick my commits?

**Comment by dstandish** [comment]
> Yeah I just saw, should I create a temp branch and cherry pick my commits?

any strategy that works is fine :) 

and yeah that sounds like a working strategy.

File: airflow/cli/commands/connection_command.py [hunk 1]
```diff
@@ -269,7 +269,7 @@ def connections_add(args):
             msg = msg.format(
                 conn_id=new_conn.conn_id,
                 uri=args.conn_uri
-                or urlunparse(
+                or urlunsplit(
```
Comment (uranusjr) [inline airflow/cli/commands/connection_command.py:268]: I don’t think this works, `urlunparse` takes a six-item iterable, but `urlunsplit` takes a five-item one. More review is needed.

Comment (westonkl) [reply airflow/cli/commands/connection_command.py:268]: I fixed this now

**Review by uranusjr — APPROVED** [review]

**Comment by WestonKing-Leatham** [comment]
> looks like something went wrong with your git operations. you have a lot of unrelated changes.

This was fixed, could you review again?

**Comment by eladkal** [comment]
Needs rebase and resolving conflicts

**Review by eladkal — APPROVED** [review]

**Pull request merged**